airflow-core/src/airflow/api_fastapi/core_api/routes/public/hitl_shared_links.py

```diff
@@ -0,0 +1,259 @@
+
+log = structlog.get_logger(__name__)
+
+
+@hitl_shared_link_endpoint_enabled
+@hitl_shared_links_router.post(
+    "/generate/{dag_id}/{dag_run_id}/{task_id}",
+    status_code=status.HTTP_201_CREATED,
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+    dependencies=[
+        Depends(requires_access_dag(method="POST", access_entity=DagAccessEntity.HITL_DETAIL)),
+    ],
+)
+def generate_shared_link(
+    # task instance identifier
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str,
+    # payload
+    generate_shared_link_request: GenerateHITLSharedLinkRequest,
+    # utitliy
+    request: Request,
+    session: SessionDep,
+) -> HITLSharedLinkResponse:
+    """
+    Generate a shared link for a Human-in-the-loop task instance.
+
+    This endpoint generates a secure, time-limited shared link that allows users
+    to interact with the Human-in-the-loop task instance. The link
+    can be configured for either respond directly or UI redirection.
+
+    :param dag_id: The dag identifier
+    :param dag_run_id: The dag run identifier
+    :param task_id: The task identifier
+    :param generate_shared_link_request: Request containing link configuration
+    :param http_request: HTTP request for base URL extraction
+    :param session: Database session for data persistence
+
+    :raises HTTPException: 400 if link generation fails due to invalid parameters
+    :raises HTTPException: 403 if Human-in-the-loop link sharing feature is not enabled
+    :raises HTTPException: 404 if the task instance does not exist
+
+    :return: SharedLinkResponse containing the generated link URL and metadata
+    """
+    return generate_hitl_shared_link_from_request(
+        # task instance identifier
+        dag_id=dag_id,
+        dag_run_id=dag_run_id,
+        task_id=task_id,
+        map_index=None,
+        # payload
+        generate_shared_link_request=generate_shared_link_request,
+        # utitliy
+        request=request,
+        session=session,
+    )
+
+
+@hitl_shared_link_endpoint_enabled
+@hitl_shared_links_router.post(
+    "/generate/{dag_id}/{dag_run_id}/{task_id}/{map_index}",
+    status_code=status.HTTP_201_CREATED,
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+    dependencies=[
+        Depends(requires_access_dag(method="POST", access_entity=DagAccessEntity.HITL_DETAIL)),
+    ],
+)
+def generate_mapped_ti_shared_link(
+    # task instance identifier
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str,
+    map_index: int,
+    # payload
+    generate_shared_link_request: GenerateHITLSharedLinkRequest,
+    # utitliy
+    request: Request,
+    session: SessionDep,
+) -> HITLSharedLinkResponse:
```

Comment:
Having two methods that are otherwise identical apart from the map_index parameter they pass on seems overkill -- we already have an established pattern of `-1` or None = not mapped, so I don't think we need two functions.